packages/component-utilities/src/getCompletedData.js

```diff
@@ -1,132 +1,83 @@
-import { tidy, complete } from '@tidyjs/tidy';
+import { tidy, complete, mutate } from '@tidyjs/tidy';
 import getDistinctValues from './getDistinctValues';
-
-function getDiffs(arr) {
-	var diffs = [];
-	for (var i = 1; i < arr.length; i++) diffs.push(arr[i] - arr[i - 1]);
-	return diffs;
-}
-
-function gcd(a, b) {
-	// Get greatest common divisor of the differences between values
-	if (a < b) return gcd(b, a);
-
-	// base case
-	if (Math.abs(b) < 0.001) return a;
-	else return gcd(b, a - Math.floor(a / b) * b);
-}
-
-function extent(values, valueof) {
-	let min;
-	let max;
-	if (valueof === undefined) {
-		for (const value of values) {
-			if (value != null) {
-				if (min === undefined) {
-					if (value >= value) min = max = value;
-				} else {
-					if (min > value) min = value;
-					if (max < value) max = value;
-				}
-			}
-		}
-	} else {
-		let index = -1;
-		for (let value of values) {
-			if ((value = valueof(value, ++index, values)) != null) {
-				if (min === undefined) {
-					if (value >= value) min = max = value;
-				} else {
-					if (min > value) min = value;
-					if (max < value) max = value;
-				}
-			}
+import { findInterval, vectorSeq } from './helpers/getCompletedData.helpers.js';
+
+/**
+ * This function fills missing data points in the given data array for a specific series.
+ *
+ * @param {Array<object>} data - The data as an array of objects.
+ * @param {string} x - The property used as x-axis.
+ * @param {string} y - The property used as y-axis.
+ * @param {string} series - The specific series in the data to be filled.
+ * @param {boolean} [nullsZero=false] - A flag indicating whether nulls should be replaced with zero.
+ * @param {boolean} [fillX=false] - A flag indicating whether the x-axis values should be filled (based on the found interval distance).
+ * @return {Array<object>} An array containing the filled data objects.
```

Comment:
```suggestion
 * @param {Record<string, unknown>[]} data - The data as an array of objects.
 * @param {string} x - The property used as x-axis.
 * @param {string} y - The property used as y-axis.
 * @param {string} series - The specific series in the data to be filled.
 * @param {boolean} [nullsZero=false] - A flag indicating whether nulls should be replaced with zero.
 * @param {boolean} [fillX=false] - A flag indicating whether the x-axis values should be filled (based on the found interval distance).
 * @return {Record<string, unknown>[]} An array containing the filled data objects.
```
`object` will make Typescript whine if you access keys on the object